### PR TITLE
Taxii Feed Fix

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
- [x] Ready

## Description
Fixes a problem where instance variables were uninitialized in the `init` method of the `TAXIIClient` class, causing an `AttributeError` to be raised when making requests.

## Minimum version of Demisto
- [x] 5.5.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by avidan-H** [comment]
@Itay4 you can see the unit-test I added in my first commit fails in the associated build.

**Comment by avidan-H** [comment]
@kirbles19 can you please take a look at the CHANGELOG file linked [here](https://github.com/demisto/content/edit/taxii-feed-fix/Integrations/FeedTAXII/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F6120)?

**Comment by kirbles19** [comment]
@avidan-H Done, no change.

File: Integrations/FeedTAXII/FeedTAXII_test.py [hunk 1]
```diff
@@ -95,3 +96,13 @@ def test_fetch_indicators(self, mocker):
             with open('FeedTAXII_test/TestCommands/indicators_results.json', 'r') as exp_f:
                 expected = json.load(exp_f)
                 assert res == expected
+
+
+class TestTaxiiClient:
+    # verify that unassigned instance variables are still instantiated
+    def test__send_request(self):
```
Comment (Itay4) [inline Integrations/FeedTAXII/FeedTAXII_test.py]:
```suggestion
    def test_send_request(self):
```

File: Integrations/FeedTAXII/FeedTAXII_test.py [hunk 2]
```diff
@@ -95,3 +96,13 @@ def test_fetch_indicators(self, mocker):
...
+        from FeedTAXII import TAXIIClient
+        client = TAXIIClient()
+        with requests_mock.Mocker() as m:
+            m.post('http://www.example.com', text='response does not matter')
```
Comment (Itay4) [inline Integrations/FeedTAXII/FeedTAXII_test.py]: lets also add a test which asserts the right auth params are initialized

File: Integrations/FeedTAXII/FeedTAXII.py [hunk 3]
```diff
@@ -523,7 +526,7 @@ def _send_request(self, url, headers, data, stream=False):
             data=data
         )
 
-        if self.username is not None and self.password is not None:
+        if self.username and self.password:
```
Comment (Itay4) [inline Integrations/FeedTAXII/FeedTAXII.py]: shouldnt be `elif` of the prev condition ?

File: Integrations/FeedTAXII/FeedTAXII.py [hunk 4]
```diff
@@ -503,16 +503,19 @@ def __init__(self, insecure: bool = True, polling_timeout: int = 20, initial_int
...
-        if credentials:
-            if '_header:' in credentials.get('identifier', None):
-                self.api_header = credentials.get('identifier', None).split('_header:')[1]
-                self.api_key = credentials.get('password', None)
-            else:
-                self.username = credentials.get('identifier', None)
-                self.password = credentials.get('password', None)
+        if not credentials:
+            credentials = {}
+        self.api_head, self.api_key, self.username, self.password = '', '', '', ''
```
Comment (Itay4) [inline Integrations/FeedTAXII/FeedTAXII.py]: i prefer we do it in the condition
so in the condition we will init `username` and `password` to be empty
in the `else` clause we will init `api_header` and `api_key`

Comment (avidan-H) [reply Integrations/FeedTAXII/FeedTAXII.py]: Unless I'm misunderstanding you - that still seems problematic - you could still have an `AttributeError` occur. Instance variables should be initialized in all cases, even if it's just to the empty string. I don't think we should be defining instance attributes on the fly. If I get an instance of a class, regardless of when or under what condition, there should be a well-defined list of instance attributes for that class, whether or not those instance attributes have non-null/0/empty-string values assigned to them or not, the _attributes_ themselves should always exist.

File: Integrations/FeedTAXII/FeedTAXII_test.py (hunk 2 above)
Comment (avidan-H) [reply Integrations/FeedTAXII/FeedTAXII_test.py]: I think they should be initialized in all cases.

Comment (avidan-H) [reply Integrations/FeedTAXII/FeedTAXII_test.py]: ... so I don't think this makes sense

**Comment by avidan-H** [comment]
Closing and reopening in a new branch based off of the purged master.

**Pull request closed without merging**